src/compiler/checker.ts

```diff
@@ -16009,13 +16009,14 @@ namespace ts {
                     unmatchedProperty.valueDeclaration
                     && isNamedDeclaration(unmatchedProperty.valueDeclaration)
                     && isPrivateIdentifier(unmatchedProperty.valueDeclaration.name)
+                    && source.symbol.valueDeclaration
                     && isClassDeclaration(source.symbol.valueDeclaration)

```

Comment:
this would be better instead:

```suggestion
                    && source.symbol.flags & SymbolFlags.Class
```

Reply:
The problem with that is that it won't know that there is a `unmatchedProperty.valueDeclaration.name` on the next line...?